### DISCUSSION TITLE
Will we need a tB StructLayoutAttribute?

### DISCUSSION THREAD

**Discussion by mburns08109** [twinbasic/twinbasic#1173] [Q&A]
In reading up about memory usage of various types, I ran across discussion about memory layout of UDTs here: https://learn.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.structlayoutattribute?view=net-6.0

I wondered that since we're doing both 32-bit and 64-bit and anticipate other cross-platform development in the future, will we be needing an equivalent StructLayoutAttribute for tB to help manage the memory layout of UDTs?

**Comment by fafalone** [comment]
There's already a `[ PackingAlignment(n) ] ` option that takes care of 99% of the instances where you need to manually change the alignment. I did run into a case of the other 1% recently; some kernel mode stuff uses a different type of alignment where e.g. `_GENERAL_LOOKASIDE` is cache aligned, and alignment and packing seem to actually be different things... I have to append 60 bytes on the end of it. Can't say I've seen that outside of kernel mode though.
